arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -287,12 +273,52 @@ class init_cameras_landmarks::priv
     return SUCCESS;
   }
 
+  kv::config_block_sptr default_config()
+  {
+    auto config = p->find_configuration("applets/init_cameras_landmarks.conf");
+
+    // choose video or image list reader based on file extension
+    auto vr_config = config->subblock_view("video_reader");
+    if (ST::GetFilenameLastExtension(video_file) == ".txt")
```

Comment:
Perhaps I'm missing something, but the way I see it `video_file` is unset until `load_sfm_constraint` is called, which appears to happen after the only time `default_configs` is called.

Reply:
Good catch!  Yes, it looks like I need to reorder some thing here so that this part of the config is done after we set the video file name.

Reply:
My [recent internal commit](https://kwgitlab.kitware.com/computer-vision/KWIVER/-/merge_requests/246/diffs?commit_id=33743a79ba01a2c395d7b0caca317d01c5d6b393) appears to work, but I don't know if this is a good approach.